### DISCUSSION TITLE
Supervisor Node

### DISCUSSION THREAD

**Discussion by cyborgdennett** [cyberbotics/webots_ros2#801] [General]
Hello, I would like to move my bot with a supervisor node and ROS2, I tried to make a webots_ros2 plugin that did a ```from controller import Supervisor```, but this did not work, it said there is already a Robot there.

So my question is: is it possible to have a ROS2 controller also be a supervisor or is it only for Controllers?

**Comment by ygoumaz** [comment]
First of all, to make things clear, the optional Ros2Supervisor node that can be added to any ROS 2 simulation has nothing to do with making a Supervisor Robot. This optional node can be used to spawn PROTOs and URDFs or publish additional information, like the `/clock` topic.

Now if you want to add a Robot with Supervisor rights, it is totally possible. To do this, you should make sure that the Supervisor field of your robot in the .wbt world file is set to TRUE. You can then create a plugin (Python or C++) and call Supervisor API functions. For C++ plugins you should add the <supervisor.h> header. For Python you can simply call on the `webots_node.robot` object.

**Reply by cyborgdennett** [reply]
Thank you this was not clear to me after browsing the docs
